protos/sigstore_common.proto

```diff
@@ -158,15 +158,24 @@ message X509Certificate {
 }
 
 enum SubjectAlternativeNameType {
-        DNS = 0;
-        Email = 1;
+        SUBJECT_ALTERNATIVE_NAME_TYPE_UNSPECIFIED = 0;
+        EMAIL = 1;
         URI = 2;
+        // OID 1.3.6.1.4.1.57264.1.7
+        // See https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#1361415726417--othername-san
+        // for more details.
+        OTHER_NAME = 3;
 }
 
 message SubjectAlternativeName {
         SubjectAlternativeNameType type = 1;
-        // A regular expression describing the expected value for the SAN.
-        string value = 2;
+        oneof identity {
+                // A regular expression describing the expected value for
+                // the SAN.
+                string regexp = 2;
```

Comment:
Is "regexp" portable across implementations?
For instance, what is the intended regexp flavour?

Reply:
This is not expected to be portable right now, I'll create an issue on fixing this as we are thinking about adding more support for regular expressions.